designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/AllDeployedScenarioService.scala

```diff
@@ -0,0 +1,92 @@
+import pl.touk.nussknacker.ui.process.fragment.{DefaultFragmentRepository, FragmentResolver}
+import pl.touk.nussknacker.ui.process.processingtypedata.{ProcessingTypeDataProvider, ValueWithPermission}
+import pl.touk.nussknacker.ui.process.repository._
+import pl.touk.nussknacker.ui.security.api.{LoggedUser, NussknackerInternalUser}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+// This class is extracted from DeploymentService to avoid a cyclic dependency:
+// DeploymentService -> EmbeddedDeploymentManager -> ... -> DeploymentService.getDeployedScenarios
+class AllDeployedScenarioService(
+    processRepository: FetchingProcessRepository[DB],
+    dbioRunner: DBIOActionRunner,
+    scenarioResolver: ScenarioResolver,
+    processingType: ProcessingType
+) extends LazyLogging {
+
+  def getDeployedScenarios(implicit ec: ExecutionContext): Future[List[DeployedScenarioData]] = {
+    implicit val userFetchingDataFromRepository: LoggedUser = NussknackerInternalUser.instance
+    for {
+      deployedProcesses <- {
+        dbioRunner.run(
+          processRepository.fetchLatestProcessesDetails[CanonicalProcess](
+            ScenarioQuery(
+              isFragment = Some(false),
+              isArchived = Some(false),
+              isDeployed = Some(true),
+              processingTypes = Some(Seq(processingType))
+            )
+          )
+        )
+      }
+      dataList <- Future.sequence(deployedProcesses.flatMap { details =>
+        val lastDeployAction = details.lastDeployedAction.get
+        // TODO: what should be in name?
+        val deployingUser  = User(lastDeployAction.user, lastDeployAction.user)
+        val deploymentData = DeploymentData(DeploymentId.fromActionId(lastDeployAction.id), deployingUser, Map.empty)
+        val deployedScenarioDataTry =
+          scenarioResolver.resolveScenario(details.json).map { resolvedScenario =>
+            DeployedScenarioData(
+              details.toEngineProcessVersion.copy(versionId = lastDeployAction.processVersionId),
+              deploymentData,
+              resolvedScenario
+            )
+          }
+        deployedScenarioDataTry match {
+          case Failure(exception) =>
+            logger.error(s"Exception during resolving deployed scenario ${details.name}", exception)
+            None
+          case Success(value) => Some(Future.successful(value))
+        }
+      })
+    } yield dataList
+  }
+
+}
+
+object AllDeployedScenarioService {
+
+  // This factory method prepare objects that are also prepared by AkkaHttpBasedRouteProvider
+  // but without dependency to ModelData - it necessary to avoid a cyclic dependency:
+  // DeploymentService -> EmbeddedDeploymentManager -> ... -> DeploymentService.getDeployedScenarios
+  def apply(dbRef: DbRef, processingType: ProcessingType)(implicit ec: ExecutionContext): AllDeployedScenarioService = {
+    val dbioRunner = DBIOActionRunner(dbRef)
```

Comment:
Does this runner need to be initialized here? Don't know the topic that much, but why cant we reuse it like in `NotificationServiceImpl`?

Reply:
It is an lightweight object, I could pass it but then we would have to params: dbref and runner, I want to keep it simple, because I'm not sure if this solution will stay it with us for a longer time - it is a kind of work around